src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.cs

```diff
@@ -46,33 +47,75 @@ internal partial class X11XamlRootHost : IXamlRootHost
 
 	public X11Window X11Window => _x11Window!.Value;
 
-	public X11XamlRootHost(Window winUIWindow, Action<Size> resizeCallback, Action closeCallback, Action<bool> focusCallback, Action<bool> visibilityCallback)
+	public X11XamlRootHost(Window winUIWindow, Action<Size> resizeCallback, Action closingCallback, Action<bool> focusCallback, Action<bool> visibilityCallback)
 	{
 		_window = winUIWindow;
 
 		_resizeCallback = resizeCallback;
-		_closeCallback = closeCallback;
+		_closingCallback = closingCallback;
 		_focusCallback = focusCallback;
 		_visibilityCallback = visibilityCallback;
 
 		_applicationView = ApplicationView.GetForWindowId(winUIWindow.AppWindow.Id);
 		_applicationView.PropertyChanged += OnApplicationViewPropertyChanged;
+		var coreApplicationView = CoreApplication.GetCurrentView();
+		coreApplicationView.TitleBar.ExtendViewIntoTitleBarChanged += UpdateWindowPropertiesFromCoreApplication;
 
 		_closed = new TaskCompletionSource();
 		Closed = _closed.Task;
-		Closed.ContinueWith(_ => _applicationView.PropertyChanged -= OnApplicationViewPropertyChanged);
+		Closed.ContinueWith(_ =>
+		{
+			_applicationView.PropertyChanged -= OnApplicationViewPropertyChanged;
+			coreApplicationView.TitleBar.ExtendViewIntoTitleBarChanged -= UpdateWindowPropertiesFromCoreApplication;
+		});
 
 		Initialize();
 		UpdateWindowPropertiesFromPackage();
+		OnApplicationViewPropertyChanged(this, new PropertyChangedEventArgs(null));
 	}
 
 	public Task Closed { get; }
 
 	private void OnApplicationViewPropertyChanged(object? sender, PropertyChangedEventArgs e)
 	{
-		// We could use _NET_WM_NAME instead
+		// We could use _NET_WM_NAME instead, which takes priority over the name set by XStoreName if both are set.
 		using var _ = X11Helper.XLock(X11Window.Display);
 		var __ = XLib.XStoreName(X11Window.Display, X11Window.Window, _applicationView.Title);
+
+		var minSize = _applicationView.PreferredMinSize;
+
+		if (minSize != Size.Empty)
+		{
+			var hints = new XSizeHints
+			{
+				flags = (int)XSizeHintsFlags.PMinSize,
+				min_width = (int)minSize.Width,
+				min_height = (int)minSize.Height
+			};
+
+			XLib.XSetWMNormalHints(X11Window.Display, X11Window.Window, ref hints);
+		}
+	}
+
+	internal void UpdateWindowPropertiesFromCoreApplication()
+	{
+		var coreApplicationView = CoreApplication.GetCurrentView();
+
+		// Sadly, there is no de jure standard for this. It's basically a set of hints used
+		// in the old Motif WM. Other WMs started using it and it became a thing.
+		var hintsAtom = X11Helper.GetAtom(X11Window.Display, X11Helper._MOTIF_WM_HINTS);
+		if (hintsAtom != IntPtr.Zero)
+		{
+			var _ = XLib.XChangeProperty(
```

Comment:
What's the return value here? Could it indicate an error when setting those properties?

Reply:
I've given up on trying to make any use of the return values from the xlib calls. Sometimes the values straight up contradict the docs, other times they don't really mean anything, most likely because the calls always "succeed" i.e. the socket write succeeds, even though it might fail on a logical level. For `XChangeProperty`, the spec says that actual_type is set to `None` if the property doesn't exist. Good call, I've previously already started to check for `None`, will add more.